core/src/main/java/org/radargun/stats/Statistics.java

```diff
@@ -173,9 +175,59 @@ default Statistics with(Statistics otherStats) {
     */
    <T> T getRepresentation(String operation, Class<T> clazz, Object... args);
 
+   StreamOfRequest getStreamOfRequest();
+
    class Converter extends ReflexiveConverters.ObjectConverter {
       public Converter() {
          super(Statistics.class);
       }
    }
+
+   class UniformRateLimiter {
+      long start = Long.MIN_VALUE;
+      final long intervalNs;
+      final AtomicLong opIndex = new AtomicLong();
+
+      UniformRateLimiter(int opsPerSec) {
+         intervalNs = 1000000000 / opsPerSec;
+      }
+
+      /* @return intended start time in ns for the operation */
+      long acquire() {
+         long currOpIndex = opIndex.getAndAdd(1);
+         return start + currOpIndex * intervalNs;
+      }
+
+      void start() {
+         start = System.nanoTime();
+      }
+   }
+
+   class StreamOfRequest {
+      private final UniformRateLimiter rateLimiter;
+
+      public StreamOfRequest(UniformRateLimiter rateLimiter) {
+         this.rateLimiter = rateLimiter;
+      }
+
+      Request nextRequest(Statistics statistics) {
+
+         Request req = new Request(statistics);
+
+         if (rateLimiter != null) {
+            long intendedTime = rateLimiter.acquire();
+            req.setIntendedStartNs(intendedTime);
+            long now;
+            while ((now = System.nanoTime()) < intendedTime)
```

Comment:
The purpose of `Statistics` is recording data. While I would be ok with adding the intended start to the request, `nextRequest` must not block the invocation - move this to the benchmark stage.

It might be better to move the `StreamOfRequest` stuff to benchmark stages completely, not sure ATM.